src/sensors/trilateration.cpp

```diff
@@ -0,0 +1,353 @@
+    }
+
+    /*
+     * The current position readings are all valid and ordered by confidence.
+     * Add them to a position message.
+     */
+    robosub::PositionsStamped position_msg;
+    position_msg.header.stamp = ros::Time::now();
+    for (unsigned int i = 0; i < position_readings.size(); ++i)
+    {
+        position_msg.distances.push_back(distance_readings[i]);
+
+        geometry_msgs::Vector3 position;
+        position.x = position_readings[i][0];
+        position.y = position_readings[i][1];
+        position.z = position_readings[i][2];
+        position_msg.positions.push_back(position);
+    }
+    position_pub.publish(position_msg);
+}
+
+/**
+ * Main entry point into the program.
+ *
+ * @param argc The number of arguments passed to ROS.
+ * @param argv A pointer to a list of arguments passed to ROS.
+ *
+ */
+int main(int argc, char **argv)
+{
+    /*
+     * Initialize the ROS node as the hydrophones node and load the
+     * hydrophone position parameters.
+     */
+    ros::init(argc, argv, "hydrophones");
+
+    ros::NodeHandle nh;
+
+    if(ros::param::getCached("hydrophones/positions/delta", delta) == false)
+    {
+        ROS_FATAL("Failed to load X hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/positions/epsilon", epsilon) ==
+            false)
+    {
+        ROS_FATAL("Failed to load Y hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/positions/zeta", zeta) == false)
+    {
+        ROS_FATAL("Failed to load Y hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/reference/x",
+                reference_hydrophone_position[0]) == false)
+    {
+        ROS_FATAL("Failed to load reference hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/reference/y",
+                reference_hydrophone_position[1]) == false)
+    {
+        ROS_FATAL("Failed to load reference hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/reference/z",
+                reference_hydrophone_position[2]) == false)
+    {
+        ROS_FATAL("Failed to load reference hydrophone position.");
+        return -1;
+    }
+
+    if (ros::param::getCached("hydrophones/pinger/depth", pinger_depth) ==
+            false)
+    {
+        ROS_FATAL("Failed to load the pinger depth.");
+        return -1;
+    }
+
+    /*
+     * Subscribe to the time delta message and publish a derived position
+     * topic.
+     */
+    ros::Subscriber delta_sub = nh.subscribe("hydrophone/30khz/delta", 1,
+            deltaCallback);
+    ros::Subscriber orientation_sub = nh.subscribe("orientation", 1,
+            orientationCallback);
+    ros::Subscriber depth_sub = nh.subscribe("depth", 1,
+            depthCallback);
+    position_pub =
+        nh.advertise<robosub::PositionsStamped>("hydrophones/position", 1);
```

Comment:
I would think hydrophones because there is more than one sensor used. Practically, either would work.